main.go

```diff
@@ -3,29 +3,97 @@
 package main
 
 import (
+	"context"
 	"errors"
 	"log"
+	"net"
+	"net/http"
 	"os"
 	"os/signal"
 	"syscall"
+	"time"
 
 	server "github.com/couchbase-examples/wasmcloud-provider-couchbase/bindings"
 	"github.com/couchbase/gocb/v2"
 	"github.com/wasmCloud/provider-sdk-go"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
 )
 
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		w.WriteHeader(http.StatusOK)
+		log.Default().Println("Health check successful")
+		// span.AddEvent("Health check successful")
+	} else {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		log.Default().Println("Health check failed")
+		// span.AddEvent("Health check failed")
+	}
+}
+
 func main() {
 	if err := run(); err != nil {
 		log.Fatal(err)
 	}
 }
 
+func newHTTPHandler() http.Handler {
+	mux := http.NewServeMux()
+	handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request)) {
+		handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
+		mux.Handle(pattern, handler)
+	}
+
+	// Register handler for healthcheck.
+	handleFunc("/health", healthCheckHandler)
+
+	// Add HTTP instrumentation for the whole server.
+	handler := otelhttp.NewHandler(mux, "/")
+	return handler
+}
+
 func run() error {
+	// Handle SIGINT (CTRL+C) gracefully.
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	// Set up OpenTelemetry.
+	otelShutdown, err := setupOTelSDK(ctx)
+	if err != nil {
+		return err
+	}
+	// Handle shutdown properly so nothing leaks.
+	defer func() {
+		err = errors.Join(err, otelShutdown(context.Background()))
+	}()
+
+	// Start HTTP server.
+	srv := &http.Server{
+		Addr:         ":8080",
+		BaseContext:  func(_ net.Listener) context.Context { return ctx },
+		ReadTimeout:  time.Second,
+		WriteTimeout: 10 * time.Second,
+		Handler:      newHTTPHandler(),
+	}
```

Comment:
I have some concerns about also using an HTTP server here for the health check, the provider does already have a health endpoint that's accessible over NATS. Is this health endpoint primarily used to demonstrate a trace or is it meant to be an indicator of provider health?

At the very least I think we should change this to a different port than `8080` since it's so common to use that for our HTTP endpoints for wasmCloud apps

Reply:
The purpose of this endpoint was to demonstrate that the trace is working when integrating via Jaeger . I am ok to change it . Do you think its better to remove this http endpoint ?

Reply:
I updated the port and simplified the tracing to replicate how it will be executed from the GET, PUT and other functions.

Reply:
I think it would be best to remove it and use the wasmCloud NATS health check endpoint for testing tracing if that's possible, that should be possible with the `handleHealthCheck` option with the provider SDK

Reply:
Ok . I deleted the http health check endpoint. Have already verified that tracing with otel setup works here.